mailpoet/lib/Newsletter/Scheduler/AutomaticEmailScheduler.php

```diff
@@ -19,12 +19,15 @@ public function __construct(
     $this->wp = $wp;
   }
 
-  public function scheduleAutomaticEmail($group, $event, $schedulingCondition = false, $subscriberId = false, $meta = false) {
+  public function scheduleAutomaticEmail($group, $event, $schedulingCondition = false, $subscriberId = false, $meta = false, $metaModifier = null) {
     $newsletters = Scheduler::getNewsletters(Newsletter::TYPE_AUTOMATIC, $group);
     if (empty($newsletters)) return false;
     foreach ($newsletters as $newsletter) {
       if ($newsletter->event !== $event) continue;
       if (is_callable($schedulingCondition) && !$schedulingCondition($newsletter)) continue;
+      if (is_callable($metaModifier)) {
```

Comment:
Having another callback seems kind of clunky but the meta data we want to store is based on the specific newsletter, and this was the cleanest approach I could come up with.

I also thought about adding a filter, something like the following:

`$meta = WPFunctions::get()->applyFilters($meta, $newsletter, $group, $event, etc)`

The individual events could then hook into the filter as part of their `init` methods.

I didn't know if we would want to expose a filter for a piece of functionality like this that probably wouldn't make sense to have other themes/plugins modifying.

I'm very open to other ideas for how to handle this and welcome suggestions.

Reply:
You're right John, it does seem clunky, but since the original `$meta` parameter was probably implied to be newsletter-agnostic, that is a viable way to tailor metadata to a specific newsletter without larger-scale refactoring of the event system. 

To make it easier to grasp, maybe it's worth adding a comment explaining why `$metaModifier` was needed, e.g.:
> The $meta argument is shared between all newsletters, to make it newsletter-specific, the $metaModifier callback can be passed.

Reply:
I definitely agree a comment is in order here!